aiosql/adapters/aiosqlite.py

```diff
@@ -37,6 +37,17 @@ async def select_one(conn, _query_name, sql, parameters, record_class=None):
                 result = record_class(**dict(zip(column_names, result)))
         return result
 
+    @staticmethod
+    async def select_value(conn, _query_name, sql, parameters, record_class=None):
+        async with conn.execute(sql, parameters) as cur:
+            result = await cur.fetchone()
```

Comment:
Is there a case in which we expect this value to be the returned result?
Would that be when record_class is not None and the result is None?

What do you think about the following? Is it doing the same thing but with one branch instead of two?

```python
result = await cur.fetchone()
result = result[0] if result else None
if record_class is not None:
    # OMG I forgot how weird cur.description is: https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.description
    column_name = cur.description[0][0]
    result = record_class(**{column_name: result})
return result
```

If this affects the approach you might take with other driver types, apply it as it makes sense.

Reply:
The code in `select_one` will return `None` if the query returns no rows even if a record class is set, right? That's what I tried to mimic in my `select_value` implementation. Your alternative is sleeker, for sure but an empty query will result in a record class instance with the column value set to `None` instead. That'd be a break in consistency again and for this instance I'd be strongly against that.

Reply:
> but an empty query will result in a record class instance with the column value set to None instead

Great point, thank you. Does make me feel like a class with a single value makes even less sense though.

This is too strange...

```python
result = await cur.fetchone()
result = result[0] if result else None
if record_class is not None:
    # OMG I forgot how weird cur.description is: https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.description
    column_name = cur.description[0][0]
    if result:
        result = record_class(**{column_name: result})
return result
```

I think the best thing is:

```python
result = await cur.fetchone()
return result[0] if result else None
```

Especially when I think about what I want for the future of the `record_class` feature.